modules/passkey/contracts/interfaces/ICustomSignerFactory.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0 <0.9.0;
+
+/**
+ * @title ICustomECSignerFactory
+ * @dev Interface for creating and verifying ECDSA signers. This is a generalized interface that should be
+ *  compatible with curves of any order size. Currently not used in the project and exists here for reference.
+ */
+interface ICustomECSignerFactory {
+    /**
+     * @notice Gets the unique signer address for the specified data.
+     * @dev The signer address should be unique for the given data. The signer does not need to be created yet.
+     * @param data The signer-specific data.
+     * @return signer The signer address.
+     */
+    function getSigner(bytes memory data) external view returns (address signer);
+
+    /**
+     * @notice Creates a new unique signer for the specified data.
+     * @dev The signer address must be unique for the given data. This should not revert if the signer already exists.
+     * @param data The signer-specific data.
+     * @return signer The signer address.
+     */
+    function createSigner(bytes memory data) external returns (address signer);
+
+    /**
+     * @notice Verifies a signature for the specified address without deploying it.
+     * @dev This should be equivalent to first deploying the signer with the factory, and then verifying the signature
+     * with it directly: `factory.createSigner(signerData).isValidSignature(message, signature)`
+     * @param message The signed message.
+     * @param signature The signature bytes.
+     * @param signerData The signer data to verify the signature for.
+     * @return magicValue Returns the legacy EIP-1271 magic value (`bytes4(keccak256("isValidSignature(bytes,bytes)")`) when the signature is valid. Reverting or returning any other value implies an invalid signature.
+     */
+    function isValidSignatureForSigner(
+        bytes32 message,
+        bytes calldata signature,
+        bytes calldata signerData
+    ) external view returns (bytes4 magicValue);
+}
+
+/**
+ * @title ICustom256BitECSignerFactory
+ * @dev Interface for creating and verifying ECDSA signers using 256-bit elliptic curves.
+ */
+interface ICustom256BitECSignerFactory {
```

Comment:
Seeing as `ICustomECSignerFactory` isn't used at all, I would actually suggest removing it. It feels a bit like we are trying to future proof, even if there is no demand for using passkeys with algorithms other than ECDSA over the P-256 curve.

Reply:
Yeah I mentioned in the `@dev` natspec that it's just kept for the future reference. I can remove it.